### ISSUE TITLE
golangci-lint step in make fails with go1.23.0

### ISSUE THREAD

**Issue by tang-fh** [prometheus/blackbox_exporter#1281]
### Host operating system: output of `uname -a`

Linux penguin 6.6.32-02877-gde0d50d4a56c #1 SMP PREEMPT_DYNAMIC Mon, 5 Aug 2024 22:06:23 +0000 x86_64 GNU/Linux

### blackbox_exporter version: output of `blackbox_exporter --version`

```
$ ./blackbox_exporter --version
blackbox_exporter, version 0.25.0 (branch: master, revision: 3dd5dfeaabc630ca0c2ec722a07f9755159ae0dd)
  build user:       REDACTED@penguin
  build date:       20240821-09:53:12
  go version:       go1.23.0
  platform:         linux/amd64
  tags:             unknown
```

### What is the blackbox.yml module config.

N/A

### What is the prometheus.yml scrape config.

N/A

### What logging output did you get from adding `&debug=true` to the probe URL?

N/A

### What did you do that produced an error?

Run `make`

### What did you expect to see?

Build completes.

### What did you see instead?

golangci-lint crashes with error

```
make: *** [Makefile.common:172: common-lint] Killed
```

I have narrowed it down to `staticcheck`. Commenting out this line in .golangci.yml avoids this error: https://github.com/prometheus/blackbox_exporter/blob/3dd5dfeaabc630ca0c2ec722a07f9755159ae0dd/.golangci.yml#L6

I couldn't repro this directly 

Also, go1.22.6 does not have this issue.

**Comment by tang-fh** [comment]
It looks like just using a newer version of golangci-lint fixes the issue.

**Comment by tang-fh** [comment]
This was fixed by PR #1285
